iati/home/templatetags/iati_tags.py

```diff
@@ -43,3 +43,35 @@ def translation_links(context, calling_page):
     return {
         'languages': language_results,
     }
+
+
+def discover_tree_recursive(current_page, calling_page):
+    """Discover page hierarchy from the page this function is called on, to the top of the main section.
+    A recursive function that discovers children if the current page is an ancestor of the page we want to draw the hierarchy to.
+    Return a flat list of dictionaries that allows the template to draw the menu linearly, rather than hierarchically."""
```

Comment:
Suggest in format:

```
Args:
    current_page (type): Description of current_page
    calling_page (type): Description of calling_page

Returns:
    list of dict: Flat list of dictionaries (each containing information about the page) that allows the template to draw the menu linearly, rather than hierarchically